pkg/util/ec2/ec2.go

```diff
@@ -83,9 +80,5 @@ func IsDefaultHostname(hostname string) bool {
 
 // HostnameProvider gets the hostname
 func HostnameProvider(hostName string) (string, error) {
-	if ecs.IsInstance() || IsDefaultHostname(hostName) {
```

Comment:
Maybe we can change the signature of HostnameProvider(s) from
`func HostnameProvider(hostName string) (string, error)` to `func HostnameProvider() (string, error)` ?